frontend/src/app/user/service/user-role.service.ts

```diff
@@ -0,0 +1,31 @@
+import {Injectable} from '@angular/core';
+import {UsersByEmailResolverService} from './users-by-email-resolver.service';
+import {Observable} from 'rxjs/Observable';
+import {UserEmailService} from './user-email.service';
+
+@Injectable()
+export class UserRoleService {
+
+    constructor(private usersByEmailResolverService: UsersByEmailResolverService,
+                private userEmailService: UserEmailService) { }
+
+    /**
+     * determine if the current user has any role in list
+     */
+    hasRoles(list: [string]): Observable<boolean> {
+        const email = this.userEmailService.getEmail();
+        return this.usersByEmailResolverService.resolve().flatMap((response) => {
+            const users = response._embedded.users;
+            let hasARoleFromList = false;
+            users.forEach(user => {
+                if (email.match(user.email)) {
```

Comment:
Might be easier to express that you are searching for the logged in user if you used the find method on the users array. Either way, that is minor so I'll approve the changes regardless.

Example
```
    let loggedInUser = users.find(user => email.match(user.email);
    if (loggedInUser) {
      const roles = user.authorities.map(role => role.authority);
      list.forEach(role => {
        hasARoleFromList = hasARoleFromList || roles.includes(role);
      });
    }
```